portal/src/graphql/portal/TemplateLocaleManagement.tsx

```diff
@@ -32,9 +34,12 @@ interface TemplateLocaleManagementProps {
   defaultTemplateLocale: TemplateLocale;
   onTemplateLocaleSelected: TemplateLocaleUpdater;
   onDefaultTemplateLocaleSelected: TemplateLocaleUpdater;
+  pendingTemplateLocales: TemplateLocale[];
```

Comment:
I suppose `pendingTemplateLocales` is used to indicate new template locales? The interaction between locale delete & add seems rather complex.

I would suggest having a single source of truth for the templates & locales, and have just a `templateUpdates: Record<string, string | null>` in the state:
- deleting a locale means setting `templateUpdates["template/path"] = null` for all templates of the locale
- adding a locale means setting `templateUpdates["template/path"] = ""` for all known templates of the locale
- the templates are merged rom the configured templates and the `templateUpdates` state. (null content means deleted)
- the locale list would be derived (useMemo) from the template paths.
- empty content would be translated to deletion on submission.
- dismissing the locale management dialog would not make request to server.

Reply:
> I suppose pendingTemplateLocales is used to indicate new template locales?

That's correct

> dismissing the locale management dialog would not make request to server.

Any thought @louischan-oursky ? If it is OK to only send request on save (the apply button in dialog only changes screen state), then having `templateUpdates: Record<string, string | null>` in state is cleaner and simpler

Reply:
@kiootic Actually shall we fetch all template in all configured locale? I think it is too complicated to fetch templates on locale is selected if we are using `templateUpdates: Record<string, string | null>`?

Reply:
I agree. I would prefer that once the screen is loaded, no further requests are made until save/discard changes.

Reply:
Actually I run into another problem, we do not know the configured locale before fetching template @kiootic Split it into 2 query?

Reply:
I see, let's just use two queries for simplicity for now, one to fetch all resource paths, another to fetch relevant resources.